src/pipeline_manager.py

```diff
@@ -58,7 +58,8 @@ def train(pipeline_name, dev_mode):
         valid_img_ids = valid_ids_data[ID_COLUMN].tolist()
         train_img_ids = list(set(annotations[ID_COLUMN].values) - set(valid_img_ids))
     else:
-        raise NotImplementedError
+        valid_img_ids = annotations.sample(PARAMS.validation_sample_size, random_state=SEED)[ID_COLUMN].values
```

Comment:
@i008 yup exactly what I meant. Can you add this to eval function as well?

Reply:
How should we ensure we are not leaking there?

Reply:
You mean with multistage training and loading of the pretrained network multiple times?

Reply:
@i008 we sample annotations, but there are multiple annotations for one img_id, we have to group by image id first or something like that
other think I found is that when default_valid_ids == True we reduce annotations for desired classes but not valid_img_ids, so in valid set we're ending up with a lot of images without any annotaions

Reply:
Ok i will inspect that